src/javascript/Edit/engineTabs/engineTabs.utils.js

```diff
@@ -61,19 +65,46 @@ export function registerEngineTabActions(nodeData) {
         primaryNodeType.hasOrderableChildNodes
     );
 
-    const actionPrefix = 'contentEditorGWTTabAction_';
-    const actionStartPriority = 3;
-
     if (tabs && tabs.length > 0) {
+        // Filter not supported tabs
+        tabs = tabs.filter(tab => !Constants.notSupportedEngineTabs.includes(tab.id));
+
+        // Build permission check query
+        let tabsPermissionCheckFields = '';
         for (let i = 0; i < tabs.length; i++) {
             const tab = tabs[i];
-            if (!Constants.notSupportedEngineTabs.includes(tab.id) && !registry.get(actionPrefix + tab.id)) {
-                registry.addOrReplace('action', actionPrefix + tab.id, openEngineTabs, {
-                    buttonLabel: tab.title,
-                    targets: ['ContentEditorHeaderActions:' + (i + actionStartPriority)],
-                    tabs: [tab.id]
-                });
+            if (tab.requiredPermission) {
+                tabsPermissionCheckFields += `\n${tab.id}:hasPermission(permissionName:"${tab.requiredPermission}")`;
             }
         }
+
+        // Permission check query
+        client.query({
+            query: gql`
```

Comment:
Can you export the query in a dedicated file or at least in a variable at the top of this file as we usually do ?